src/rt/minfo.d

```diff
@@ -457,10 +457,15 @@ struct ModuleGroup
 
             // add this module to the construction order list
             ctors[ctoridx++] = current;
+            return true;
         }
 
-        immutable(ModuleInfo)*[] doSort(size_t relevantFlags)
+        // `cycle` is set to `true` if deprecated cycle error otherwise set `result`.
+        void doSort(size_t relevantFlags, ref bool cycle, ref immutable(ModuleInfo)*[] result)
```

Comment:
Returning both a bool and a ref with the same value strikes me as odd and a bit eyebrow raising. I'd need a better reason to use it.